### PR TITLE
Jeyanthi sm/years experience1758

### PR DESCRIPTION
**Pull request by ajaywayase**

### CODE REVIEW

**Review by jeyanthi-sm — APPROVED** [review]

**Review by jeyanthi-sm — APPROVED** [review]

File: packages/models/newApplicant.ts [hunk 1]
```diff
@@ -11,12 +11,12 @@ export interface NewApplicant {
   commitment: number;
   extraInfo?: string;
   portfolioLink?: string;
+  yearsExperience?: number;
```
Comment (Enjoy2Live) [inline packages/models/newApplicant.ts:14]: this field is nowhere to be found in the api response.

@becevka  do we plan to add this to applicants collection?

File: packages/models/newApplicant.ts [hunk 2]
```diff
@@ -11,12 +11,12 @@ export interface NewApplicant {
...
   experience: string;
   reason: string;
   accepted: boolean;
   zip: number;
   role: string;
-  //id?: string;
```
Comment (Enjoy2Live) [inline packages/models/newApplicant.ts:19]: id is part of the response, why did you comment it?

**Review by Enjoy2Live — CHANGES_REQUESTED** [review]
please go over the requested changes and let me know if you have questions or any feedback

File: packages/models/newApplicant.ts (hunk 2 above)
Comment (jeyanthi-sm) [reply packages/models/newApplicant.ts:19]: I am not able to insert the applicant details when Id was present . I get this error - 
>>>>>>>>>>>>>
[2024-06-27 07:54:27.942] error: insert into `applicants` (`accepted`, `age`, `commitment`, `created_at`, `email`, `experience`, `id`, `level`, `name`, 
`published_at`, `reason`, `role`, `updated_at`, `years_experience`, `zip`) values (false, 0, 21, '2024-06-27 07:54:27.940', 'June27@gmail.com', 'sss', '1', 'intermediate', 'June27', '2024-06-27 07:54:27.917', 'sss', 'Front-end Developer', '2024-06-27 07:54:27.940', 43, '0') returning `id` - UNIQUE constraint failed: applicants.id
SqliteError: insert into `applicants` (`accepted`, `age`, `commitment`, `created_at`, `email`, `experience`, `id`, `level`, `name`, `published_at`, `reason`, `role`, `updated_at`, `years_experience`, `zip`) values (false, 0, 21, '2024-06-27 07:54:27.940', 'June27@gmail.com', 'sss', '1', 'intermediate', 
'June27', '2024-06-27 07:54:27.917', 'sss', 'Front-end Developer', '2024-06-27 07:54:27.940', 43, '0') returning `id` - UNIQUE constraint failed: applicants.id
    at Client_BetterSQLite3._query (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\dialects\better-sqlite3\index.js:39:40)
    at executeQuery (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\execution\internal\query-executioner.js:37:17)
    at Client_BetterSQLite3.query (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\client.js:154:12)
    at Runner.query (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\execution\runner.js:141:36)
    at ensureConnectionCallback (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\execution\internal\ensure-connection-callback.js:13:17)     
    at Runner.ensureConnection (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\execution\runner.js:318:20)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Runner.run (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\execution\runner.js:30:19)
    at async Object.execute (C:\Users\meenu\devlaunchers\strapiv4\node_modules\@strapi\database\dist\index.js:4248:22)
    at async Object.create (C:\Users\meenu\devlaunchers\strapiv4\node_modules\@strapi\database\dist\index.js:5022:19)
>>>>>>>>>>>
After commenting out this line, the record got inserted in the applicant content type.

Comment (jeyanthi-sm) [reply packages/models/newApplicant.ts:19]: if I pass an empty string or a space to the field id, I get this error -
>>>>>>
 [2024-06-27 11:14:01.469] error: insert into `applicants` (`accepted`, `age`, `commitment`, `created_at`, `email`, `experience`, `id`, `level`, `name`, 
`published_at`, `reason`, `role`, `updated_at`, `years_experience`, `zip`) values (false, 0, 17, '2024-06-27 11:14:01.468', 'rt@gmail.com', 'ssss', '', 
'intermediate', 'rtrtrt', '2024-06-27 11:14:01.450', 'sss', 'Front-end Developer', '2024-06-27 11:14:01.468', 55, '0') returning `id` - datatype mismatch
SqliteError: insert into `applicants` (`accepted`, `age`, `commitment`, `created_at`, `email`, `experience`, `id`, `level`, `name`, `published_at`, `reason`, `role`, `updated_at`, `years_experience`, `zip`) values (false, 0, 17, '2024-06-27 11:14:01.468', 'rt@gmail.com', 'ssss', '', 'intermediate', 'rtrtrt', '2024-06-27 11:14:01.450', 'sss', 'Front-end Developer', '2024-06-27 11:14:01.468', 55, '0') returning `id` - datatype mismatch
    at Client_BetterSQLite3._query (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\dialects\better-sqlite3\index.js:39:40)
    at executeQuery (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\execution\internal\query-executioner.js:37:17)
    at Client_BetterSQLite3.query (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\client.js:154:12)
    at Runner.query (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\execution\runner.js:141:36)
    at ensureConnectionCallback (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\execution\internal\ensure-connection-callback.js:13:17)     
    at Runner.ensureConnection (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\execution\runner.js:318:20)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Runner.run (C:\Users\meenu\devlaunchers\strapiv4\node_modules\knex\lib\execution\runner.js:30:19)
    at async Object.execute (C:\Users\meenu\devlaunchers\strapiv4\node_modules\@strapi\database\dist\index.js:4248:22)
    at async Object.create (C:\Users\meenu\devlaunchers\strapiv4\node_modules\@strapi\database\dist\index.js:5022:19)
>>>>

Comment (jeyanthi-sm) [reply packages/models/newApplicant.ts:19]: Added the line that was removed (commented).

**Review by Enjoy2Live — APPROVED** [review]
LGTM now

**Pull request merged**